src/celeritas/ext/detail/ScopedRootErrorHandler.hh

```diff
@@ -0,0 +1,45 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2022 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/ext/detail/RootErrorHandler.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+namespace celeritas
+{
+namespace detail
+{
+
+//---------------------------------------------------------------------------//
+/*!
+ *  ROOT Error Handler for Celeritas.
+ *
+ */
+
+void RootErrorHandler(int rootlevel, bool abort_bool, const char *location,
+                          const char *msg);
+
+//---------------------------------------------------------------------------//
+/*!
+ * Install a ROOT Error Handler to redirect the message toward the
+ * Celeritas Logger.
+ *
+ */
+
+class ScopedRootErrorHandler
+{
+   decltype(&RootErrorHandler) previous_;
+
+public:
+    // Install the error handler
+    ScopedRootErrorHandler();
+
+    // Return to the previous error handler.
+    ~ScopedRootErrorHandler();
+};
+
```

Comment:
You'll need to add
```c++
#if !CELERITAS_USE_ROOT
//!@{
//! Do nothing if ROOT is disabled
inline ScopedRootErrorHandler::ScopedRootErrorHandler() {};
inline ScopedRootErrorHandler::~ScopedRootErrorHandler() {};
//!@}
#endif
```
for the case when ROOT is not enabled.
(and make sure to `#include "celeritas_config.h"` at the top).